lib/std/io/Reader.zig

```diff
@@ -326,7 +326,7 @@ pub fn isBytes(self: Self, slice: []const u8) anyerror!bool {
 
 pub fn readStruct(self: Self, comptime T: type) anyerror!T {
     // Only extern and packed structs have defined in-memory layout.
-    comptime assert(@typeInfo(T).Struct.layout != .Auto);
+    comptime assert(@typeInfo(T).Struct.layout == .Extern);
```

Comment:
why remove the ability to read packed structs?

Reply:
See #12960, I'll link it in the commit.

Reply:
the comment above should be edited as well